hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/upgrade/UpgradeDowngrade.java

```diff
@@ -130,6 +145,29 @@ public boolean needsUpgrade(HoodieTableVersion toWriteVersion) {
    * @param instantTime current instant time that should not be touched.
    */
   public void run(HoodieTableVersion toVersion, String instantTime) {
+    // Fetch version from property file and current version
+    HoodieTableVersion fromVersion = metaClient.getTableConfig().getTableVersion();
+    // Determine if we are upgrading or downgrading
+    boolean isUpgrade = fromVersion.versionCode() < toVersion.versionCode();
+    if (isUpgrade && !config.autoUpgrade()) {
+      // if we are attempting to upgrade and auto-upgrade is disabled
+      // we set the write config table version to bounded by the current hudi table version
+      // and then exit out the upgrade process
+      config.setValue(HoodieWriteConfig.WRITE_TABLE_VERSION, String.valueOf(fromVersion));
```

Comment:
if auto upgrade is disabled, we don't need to set anything here.
just log a warn msg and return right away.

Reply:
Just to confirm, I saw that on `master` when it would hit these autoUpgrade disabled cases, we would configure the write config like so.

https://github.com/apache/hudi/blob/master/hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/upgrade/SevenToEightUpgradeHandler.java#L95

https://github.com/apache/hudi/blob/master/hudi-client/hudi-client-
common/src/main/java/org/apache/hudi/table/upgrade/EightToNineUpgradeHandler.java#L53

Hence why i did the same. If this is not needed though then will remove it.
cc @yihua

Reply:
Will remove this line